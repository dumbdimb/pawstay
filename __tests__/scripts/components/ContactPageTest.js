jest.unmock('../../../src/scripts/components/ContactPage');

import React from 'react';
import renderer from 'react-test-renderer';
import { shallow } from 'enzyme';

import ContactPage from '../../../src/scripts/components/ContactPage';
import GoogleMailApiClient from '../../../src/scripts/email/GoogleMailApiClient';

describe('ContactPage', () => {
    let tree;

    beforeEach(() => {
        GoogleMailApiClient.sendMessage = jest.fn();
        tree = shallow(<ContactPage/>);
    });

    it('renders correctly', () => {
        expect(renderer.create(<ContactPage/>).toJSON()).toMatchSnapshot();
    });

    describe('the form', () => {
        it('changes state when I change the ownerName field', () => {
            const testValue = 'Bob Dole';

            const fakeEvent = {
                target: {
                    value: testValue,
                    name: 'ownerName'
                }
            };

            tree.find('.contact-form__field').at(0).props().onChange(fakeEvent);

            expect(tree.update().state('ownerName')).toEqual(testValue);
        });

        it('changes state when I change the dogName field', () => {
            const testValue = 'Spike';

            const fakeEvent = {
                target: {
                    value: testValue,
                    name: 'dogName'
                }
            };

            tree.find('.contact-form__field').at(1).props().onChange(fakeEvent);

            expect(tree.update().state('dogName')).toEqual(testValue);
        });

        it('changes state when I change the ownerEmail field', () => {
            const testValue = 'bob@dole.com';

            const fakeEvent = {
                target: {
                    value: testValue,
                    name: 'ownerEmail'
                }
            };

            tree.find('.contact-form__field').at(2).props().onChange(fakeEvent);

            expect(tree.update().state('ownerEmail')).toEqual(testValue);
        });

        it('changes state when I change the subject field', () => {
            const testValue = 'My dog needs some love and I can&apos;t give it to him!';

            const fakeEvent = {
                target: {
                    value: testValue,
                    name: 'subject'
                }
            };

            tree.find('.contact-form__field').at(3).props().onChange(fakeEvent);

            expect(tree.update().state('subject')).toEqual(testValue);
        });

        it('changes state when I change the body field', () => {
            const testValue = 'Aaaaaaaaahhhhhhhhhh help me!';

            const fakeEvent = {
                target: {
                    value: testValue,
                    name: 'body'
                }
            };

            tree.find('.contact-form__field--body').props().onChange(fakeEvent);

            expect(tree.update().state('body')).toEqual(testValue);
        });

        describe('the submit button', () => {
            it('sends an e-mail to clarissa@pawstay.dog', () => {
                tree.setState({
                    ownerName: 'Bob Dole',
                    dogName: 'Dog Dole',
                    ownerEmail: 'bob@dole.com',
                    subject: 'Please watch my dog',
                    body: 'I don\'t have any other options'
                });

                tree = tree.update();

                tree.find('.contact-form__submit').simulate('click');

                expect(GoogleMailApiClient.sendMessage).toBeCalledWith('Bob Dole', 'Dog Dole', 'bob@dole.com', 'Please watch my dog', 'I don\'t have any other options');
            });
        });
    });
});