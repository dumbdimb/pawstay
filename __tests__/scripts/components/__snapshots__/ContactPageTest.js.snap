// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`ContactPage renders correctly 1`] = `
<div
  className="page"
>
  <div
    className="contact__preheader"
  >
    even though this is a dog boarding and daycare website,
  </div>
  <div
    className="contact__header"
  >
    talk to a human
  </div>
  <div
    className="contact__subheader"
  >
    Want to ask for spot availability or make a meet and greet appointment?
    <br />
    Fill out this form below and we will get back to you soon!
  </div>
  <div
    className="contact-form"
  >
    <div
      className="contact-form__left"
    >
      <input
        className="contact-form__field"
        name="ownerName"
        onChange={[Function]}
        placeholder="*Name of the dog owner"
        type="text"
      />
      <input
        className="contact-form__field"
        name="dogName"
        onChange={[Function]}
        placeholder="*Name of the furbaby"
        type="text"
      />
      <input
        className="contact-form__field"
        name="ownerEmail"
        onChange={[Function]}
        placeholder="*Dog owner's email address"
        type="text"
      />
      <input
        className="contact-form__field"
        name="subject"
        onChange={[Function]}
        placeholder="*What's the message about"
        type="text"
      />
    </div>
    <div
      className="contact-form__right"
    >
      <textarea
        className="contact-form__field contact-form__field--body"
        name="body"
        onChange={[Function]}
        placeholder="*Your message"
      />
      <div
        className="contact-form__submit"
        onClick={[Function]}
      >
        send message
      </div>
    </div>
  </div>
  <div
    className="contact__location"
  >
    <div
      className="contact__location__pin-icon"
    />
    <div
      className="contact__location__right"
    >
      <div
        className="contact__location__info"
      >
        Palatine, IL
      </div>
      <div
        className="contact__location__info"
      >
        USA
      </div>
    </div>
  </div>
</div>
`;
